holdings_maintenance/pds4/pds4infoshelf.py

```
@@ -0,0 +1,931 @@
+
+        keys = list(shelfdict.keys())
+        keys.sort()
+        for key in keys:
+            logger.error('Shelf info for missing file', key)
+
+    except (Exception, KeyboardInterrupt) as e:
+        logger.exception(e)
+        raise
+
+    finally:
+        return logger.close()
```

Ruff (0.14.0)

`return` inside `finally` blocks cause exceptions to be silenced

(B012)